src/Symfony/Component/DependencyInjection/Dumper/GraphvizDumper.php

```diff
@@ -173,7 +173,7 @@ private function findNodes()
         foreach ($container->getServiceIds() as $id) {
             $service = $container->get($id);
 
-            if (in_array($id, array_keys($container->getAliases()))) {
+            if (array_key_exists($id, $container->getAliases())) {
```

Comment:
I double checked (was in doubt to push this or not), id/aliases are strings, so strict array_key_exists doesn't make any harm (just as you commented).